cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,252 @@
+"""
+``remove_fixable_singularities`` specifies a method to remove fixable singularities from a model
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(expr, V, sp, Vmin, Vmax):
+    """
+    Generates a new (piecewise) expression based on expr with a linear interpolation when V is between Vmin and Vmax
+
+    The returned expression between Vmin and Vmax is::
+    f(vs) + (f(ve) - f(vs)) * (V - vs) / (ve - vs)
+
+    :param: The expression to turn into a piecewise
+    :param: V the voltage variable.
+    :param: the value of the singularity point: the value of V for whih U == 0
+    :param: the value of the lower bound: the value of V for which U - U_offset == 0
+    :param: the value of the upper bound: the value of V for which U + U_offset == 0
+    :return: expr with a singularity exception around sp, if sp is not None else return the original expr
+    """
+    f_Vmin = expr.xreplace({V: Vmin})
+    f_Vmax = expr.xreplace({V: Vmax})
+    return Piecewise((f_Vmin + ((V - Vmin) / (Vmax - Vmin)) * (f_Vmax - f_Vmin),
+                      Abs(V - sp) < Abs((Vmax - Vmin) / 2)), (expr, True))
```

Comment:
why would there be a jump? It should be equivalent. 

The main reason for not going with `Vmin < V < Vmax` is that as far as I can see you can't create that in sympy (StrictLessThan takes exactly 2 parameters). Also it wouldn't render well at all in C++ `Vmin < V < Vmax`  === `(Vmin < V) < Vmax` where `(Vmin < V)` gets calculated and the 0/1 result is used to compare to Vmax

The proper alternative would be `(V < Vmax) & (Vmin < V)`
I think @mirams prefered the `fabs` version if I remember correctly?

Reply:
Yeah sure `(V < Vmax) and (Vmin < V)` is the sympy way to to write it. But what I'm saying is this is not equivalent to the abs version, because `Vmax != sp + (Vmax - Vmin) / 2` and `Vmin != sp - (Vmax - Vmin) / 2`.